test/rules/promise-async/test.ts.lint

```diff
@@ -0,0 +1,38 @@
+const nonAsyncPromiseFunction = function(p: Promise<void>) { return p; };
```

Comment:
@buu700 yeah that's fine; @nchen63 can look into the longer term solution after this PR merges